### DISCUSSION TITLE
Como crear un programa de Python con cmd.

### DISCUSSION THREAD

**Discussion by JoseOlayo** [LaunchX-InnovaccionVirtual/CursoIntroPython#95] [Q&A] [answered]
Hola, comunidad. Buenas tardes, espero que se encuentren bien,

Ojalá alguien conozca la manera de crear un archivo .py desde cmd y escribir su código desde ahí. Ya que en una kata vi que al ejecutar el programa el archivo se creaba solo, pero en mi caso. El ejecutar código Python desde cmd no me genera ningún archivo. Si alguien conoce la solución o algo que me esté faltando hacer, les agradecería mucho su respuesta. Saludos.

**Comment by Dfredude** [comment]
- Esta un poco dificil entender que es lo que quieres. Para crear un archivo .py desde cmd corre:
`echo '' >> program.py` desde el directorio donde lo quieres crear. Ejemplo de que funciona:
![image](https://user-images.githubusercontent.com/72330887/153517341-e1b5d364-2b06-4f04-95e5-88f17ada7b08.png)

-Realmente no entiendo que quieres lograr con eso, si VS Code ya te facilita tanto la creacion de archivos y edicion de texto. En cmd no puedes editar text. Pero puedes abrir VS Code con el comando `code <archivo>`
Tal ejemplo:
![image](https://user-images.githubusercontent.com/72330887/153517542-26767789-091c-40c0-8184-6dd5b1dc5301.png)
- Ya en VS Code si quieres generar un archivo **_.py_** de tu Jupyter Notebook , solo haz estos dos pasos
> 1
> ![1](https://user-images.githubusercontent.com/72330887/153518007-cddbb638-359f-4390-b6ff-624a9ad61bb4.png)
> 2
> ![image](https://user-images.githubusercontent.com/72330887/153518049-6f8a27ea-3986-4aca-b36d-92d35093673f.png)
## Espero haberte ayudado!

**Reply by JoseOlayo** [reply]
Hola, en sí solo busco aumentar un poco mis habilidades en cmd alejarme un poquito de la GUI. Además de que no sabía si me estaba faltando esa parte.
Si me ha ayuda la información que me compartiste, ahora he aprendido unas cuantas cosas más. Muchas gracias 😃

**Comment by EmmanuelMontes** [comment]
Hola,
Puedes hacer la instalación de WSL, para lo cual requieres:
1.-Agregar la característica de WSL desde agregar o quitar programas.
2.-Instala Ubuntu desde WinStore.
3.-Especificas passwd solicitado.
3.-Desde cmd te ubicas en el directorio de tus katas a editar, ahí tecleas WSL y listo ya estas en un entorno de ejecución a nivel Terminal.
4.-Abres con "vi" o con "nano" tu archivo, por si quieres editar contenido, sino si lo que deseas es ejecutar solo tipeas python3 archivo.py y ya, ves el programa corriendo.
Saludos

**Reply by JoseOlayo** [reply]
Ya implica el uso de otro SO. Pero también es muy útil saber esto.  Gracias por tu respuesta.

**Comment by 3rickDJ** [comment, marked as answer]
# Sí se puede editar y crear archivos :star_struck:! 

### Gran solución para quien quiera editar y/o crear sus archivos en la terminal es `Vim` o `Neovim`.
Neovim es editor de texto muy **flexible** y **poderoso**. Te permite `editar código desde la terminal`; es mucho muy configurable; `increíblemente veloz`, no solo en su ejecución (aún con muchos plugins) también al momento de leer o escribir código **sin despegar las manos de tu teclado**, esto lo logra a través de sus distintos `modos` ( visual, insertar, etc... ¡Una característica bárbara!).

Está disponible para linux, windows y macOS.

Puedes descargarlo [aquí](https://github.com/neovim/neovim/wiki/Installing-Neovim), con un administrador de paquetes como `chocolatey`  o desde un ejecutable.
